### PR TITLE
Add cli options "3" - do not round up the RTT time.

### PR DESCRIPTION
**Pull request by liboabo**
In original version ping, when the delay is >= 100ms the resulting time is rounded for milliseconds.

$ ping -c3 192.168.56.102
PING 192.168.56.102 (192.168.56.102) 56(84) bytes of data.
64 bytes from 192.168.56.102: icmp_seq=1 ttl=64 time=81.0 ms
64 bytes from 192.168.56.102: icmp_seq=2 ttl=64 time=130 ms
64 bytes from 192.168.56.102: icmp_seq=3 ttl=64 time=81.4 ms

To avoid this added new command line option "-3".
 
$ ping -c3 -3 192.168.56.102
PING 192.168.56.102 (192.168.56.102) 56(84) bytes of data.
64 bytes from 192.168.56.102: icmp_seq=1 ttl=64 time=138.515 ms
64 bytes from 192.168.56.102: icmp_seq=2 ttl=64 time=101.676 ms
64 bytes from 192.168.56.102: icmp_seq=3 ttl=64 time=107.434 ms

### CODE REVIEW

**Comment by pevik** [comment]
Simple question: why do you need it?

I see your `-3` behavior is similar to the default for ping from busybox, fping behaves similarly to our ping from iputils.

There are formal things which would need to be fixed (no new README file, fixing formatting - wrong indent, missing a real explanation for the reason), but the main question is whether it's really needed to be included.

File: ping/ping_common.c [hunk 1]
```diff
@@ -83,6 +83,8 @@ void usage(void)
 		"  -V                 print version and exit\n"
 		"  -w <deadline>      reply wait <deadline> in seconds\n"
 		"  -W <timeout>       time to wait for response\n"
+		"\nRRI options (Huawei):\n"
```
Comment (pevik) [inline ping/ping_common.c:86]: What `RRI` means? Why do you mention Huawei?

**Comment by pevik** [comment]
Please stash the changes to a single commit (`git rebase -i` and then use `s` for the second commit and then force push: `git push -f`). Or I'll do it during merge.

File: ping/ping.c [hunk 2]
```diff
@@ -337,6 +337,7 @@ main(int argc, char **argv)
 		.source6.sin6_family = AF_INET6,
 		.ni.query = -1,
 		.ni.subject_type = -1,
+    .rtt_precision = 0
```
Comment (pevik) [inline ping/ping.c:340]: This has wrong indent, please fix it (and other wrong indents).

File: ping/ping_common.c (hunk 1 above)
Comment (liboabo) [reply ping/ping_common.c]: The reason is to have the -A and -3 options in the same binary to see the actual RTT deviation on the interface.

**Review by pevik — CHANGES_REQUESTED** [review]
@liboabo I'd like to know why you you need that precise option.

@iputils/maintainers FYI this feature LGTM, but I'd like to know your opinion.

File: ping/ping.c (hunk 2 above)
Comment (liboabo) [reply ping/ping.c:340]: Change spaces to tab characters for indent?

Comment (liboabo) [reply ping/ping.c:340]: I will do it.

Comment (liboabo) [reply ping/ping.c]: Done

**Comment by pevik** [comment]
@liboabo Merged as 48ae5c97eb7217db59c919e9be3d8c7fcad795a8, thanks! FYI in the end I merged slightly different version (I did various fixes):

* Don't print `0.000` when time is not transferred (on `-s 15`)
* Document `-3` in the man page
* Rename opt + use bitfield
* Other cleanup

```diff
diff --git doc/ping.xml doc/ping.xml
index f4f4736..fc9b974 100644
--- doc/ping.xml
+++ doc/ping.xml
@@ -24,7 +24,7 @@ xml:id="man.ping">
     <cmdsynopsis sepchar=" ">
       <command>ping</command>
       <arg choice="opt" rep="norepeat">
-        <option>-aAbBdCDfhHLnOqrRUvV46</option>
+        <option>-aAbBdCDfhHLnOqrRUvV346</option>
       </arg>
       <arg choice="opt" rep="norepeat">
         <option>-c
@@ -127,6 +127,14 @@ xml:id="man.ping">
       <title>OPTIONS</title>
     </info>
     <variablelist remap="TP">
+      <varlistentry>
+        <term>
+          <option>-3</option>
+        </term>
+        <listitem>
+          <para>RTT precision (do not round up the result time).</para>
+        </listitem>
+      </varlistentry>
       <varlistentry>
         <term>
           <option>-4</option>
diff --git ping/ping.c ping/ping.c
index 0229db5..5588432 100644
--- ping/ping.c
+++ ping/ping.c
@@ -337,7 +337,6 @@ main(int argc, char **argv)
 		.source6.sin6_family = AF_INET6,
 		.ni.query = -1,
 		.ni.subject_type = -1,
-		.rtt_precision = 0
 	};
 	/* FIXME: global_rts will be removed in future */
 	global_rts = &rts;
@@ -372,9 +371,8 @@ main(int argc, char **argv)
 				error(2, 0, _("only one -4 or -6 option may be specified"));
 			hints.ai_family = AF_INET;
 			break;
-		/* Don't round up the rtt result */
 		case '3':
-			rts.rtt_precision = 1;
+			rts.opt_rtt_precision = 1;
 			break;
 		case 'b':
 			rts.broadcast_pings = 1;
diff --git ping/ping.h ping/ping.h
index ceb8a89..98d035d 100644
--- ping/ping.h
+++ ping/ping.h
@@ -195,7 +195,6 @@ struct ping_rts {
 	int rtt_addend;
 	uint16_t acked;
 	int pipesize;
-	int rtt_precision;
 
 	ping_func_set_st ping4_func_set;
 	ping_func_set_st ping6_func_set;
@@ -234,6 +233,7 @@ struct ping_rts {
 	unsigned int
 		opt_adaptive:1,
 		opt_audible:1,
+		opt_connect_sk:1,
 		opt_flood:1,
 		opt_flood_poll:1,
 		opt_flowinfo:1,
@@ -246,6 +246,7 @@ struct ping_rts {
 		opt_outstanding:1,
 		opt_pingfilled:1,
 		opt_ptimeofday:1,
+		opt_rtt_precision:1,
 		opt_quiet:1,
 		opt_rroute:1,
 		opt_so_debug:1,
@@ -254,8 +255,7 @@ struct ping_rts {
 		opt_strictsource:1,
 		opt_timestamp:1,
 		opt_ttl:1,
-		opt_verbose:1,
-		opt_connect_sk:1;
+		opt_verbose:1;
 };
 /* FIXME: global_rts will be removed in future */
 extern struct ping_rts *global_rts;
diff --git ping/ping_common.c ping/ping_common.c
index 7c7c3d9..ebe7888 100644
--- ping/ping_common.c
+++ ping/ping_common.c
@@ -50,6 +50,7 @@ void usage(void)
 		"  ping [options] <destination>\n"
 		"\nOptions:\n"
 		"  <destination>      DNS name or IP address\n"
+		"  -3                 RTT precision (do not round up the result time)\n"
 		"  -a                 use audible ping\n"
 		"  -A                 use adaptive ping\n"
 		"  -B                 sticky source address\n"
@@ -83,7 +84,6 @@ void usage(void)
 		"  -V                 print version and exit\n"
 		"  -w <deadline>      reply wait <deadline> in seconds\n"
 		"  -W <timeout>       time to wait for response\n"
-		"  -3                 RTT precision - don't round up the result time.\n"
 		"\nIPv4 options:\n"
 		"  -4                 use IPv4\n"
 		"  -b                 allow pinging broadcast\n"
@@ -819,8 +819,10 @@ restamp:
 			printf(_(" (truncated)\n"));
 			return 1;
 		}
-		if (rts->timing && !rts->rtt_precision) {
-			if (triptime >= 100000 - 50)
+		if (rts->timing) {
+			if (rts->opt_rtt_precision)
+				printf(_(" time=%ld.%03ld ms"), triptime / 1000, triptime % 1000);
+			else if (triptime >= 100000 - 50)
 				printf(_(" time=%ld ms"), (triptime + 500) / 1000);
 			else if (triptime >= 10000 - 5)
 				printf(_(" time=%ld.%01ld ms"), (triptime + 50) / 1000,
@@ -833,9 +835,6 @@ restamp:
 				       triptime % 1000);
 		}
 
-		if (rts->rtt_precision)
-			printf(_(" time=%ld.%03ld ms"), triptime / 1000, triptime % 1000);
-
 		if (dupflag && (!multicast || rts->opt_verbose))
 			printf(_(" (DUP!)"));
 		if (csfailed)

**Pull request closed without merging**